### DISCUSSION TITLE
Please allow duplicating a pinned image to a few same instances

### DISCUSSION THREAD

**Discussion by elad2109** [flameshot-org/flameshot#2516] [Ideas]
After pinning an image, please allow duplicating a few instances of this pinned image.

**Comment by mmahmoudian** [comment]
Would you please elaborate on the use case of such feature.

**Comment by elad2109** [comment]
Sure.

Sometimes I want to pin the same image twice and show two alternatives.
Then I take a new screenshot of the two pinned capture.

It makes it so much easier instead of the need to select the exact selection a couple of times.

A fake unreal illustrating example: 

![image](https://user-images.githubusercontent.com/1093841/161407777-1610b5b4-3e7c-40de-90c1-2c4116d57453.png)

**Reply by mmahmoudian** [reply]
But then if the dame pinned image is duplicated, the annotations would also get duplicated, so they will all look exactly identical. What you showed is the same screenshot but annotated in different ways.

**Reply by elad2109** [reply]
Two steps:

First I duplicate the pinned capture a few times. 
Then I annotate each of them differently.

I get a few alternatives compared next to each other.

**Reply by mmahmoudian** [reply]
This is not possible to annotate a screenshot after it has got pinned, so by default this particular feature request is not possible unless annotation after pinning gets implemented.

**Reply by elad2109** [reply]
I mean duplicating the same exact instance. And then capturing all of them and annotating differently. 


Can I ask when do you use pinned captures?

**Reply by mmahmoudian** [reply]
I personally use the pin when:
- I want to bring things from different pages of a PDF or multiple web pages, 
- I generate different version of the some graphs and I don't want to store them on the disk but want a quick way to compare
- want to keep some instructions at hand since pinned images are always on top and also I can change their size fast
- I want to scan a barcode from my monitor with my phone but the size on the webpage is not right, I use the pin to be able to granularly adjust the size and read the barcode 

In neither of these situations I would need multiple identical copies of the same image.

**Reply by elad2109** [reply]
Here is a perfect a usage example:
I wanted to show two options side by side.

Possible to add this feature request?

![image](https://user-images.githubusercontent.com/1093841/164109990-f79c8de5-3610-4ec8-bd82-24eee6c3e5ae.png)

**Reply by mmahmoudian** [reply]
What you want is not duplicating a pin, what you want is two images of the same are of your screen and annotate them differently. As far as I understand, in Flameshot when the annotation is done and before image is pinned, the image and annotations would get flattened and no further modifications can happen.

So what you want cannot happen in the current flow of data in Flameshot. I'm curious to know what other devs (@borgmanJeremy @veracioux) think about this scenario.

The easiest way for you would be to repeat the same screenshot. If your concern is the dimensions of the two screenshots, you can use the `--geometry` argument in the command line to specify the exact dimension and position of the selected area. If you need help for using it, let us know and we will guide you 😊

**Reply by veracioux** [reply]
We could make it so that the pin widget is itself composed of a base image and tool widgets overlaid on top of it, without baking them into a "final" image. This could be easily incorporated into the idea of allowing the pin widget to become a capture window. But this last part is not naive and will take a fair bit of time to implement.

**Comment by ElonVampire** [comment]
收到！
